src/Modules/AuthModule/components/Register/Register.tsx

```diff
@@ -1,7 +1,287 @@
+                </div>
+                <div className="row">
+                  <div className="col-md-12 text-center position-relative">
+                    <div>
+                      {selectedImage ? (
+                        <div className="">
+                          <img
+                            src={selectedImage}
+                            alt=""
+                            className="image-width rounded-circle border-0 "
+                          />
+                        </div>
+                      ) : (
+                        <div className="text-center position-relative">
+                          <img className="image-width  " src={Avatar} />
+                        </div>
+                      )}
+                      <label
+                        htmlFor="image"
+                        className="upload position-absolute"
+                      >
+                        <i className="fa-solid fa-camera fa-lg"></i>
+                      </label>
+                    </div>
 
+                    <input
+                      type="file"
+                      accept="image/jpeg , image/png , image/jpg"
+                      className="d-none"
+                      id="image"
+                      {...register("profileImage", {
+                        onChange: (e) => changingImage(e),
+                      })}
+                    />
+                  </div>
+                  <div className="col-md-6">
+                    <div className="input m-3">
+                      <label className="my-1">UserName</label>
+                      <input
+                        type="text"
+                        className="form-control p-0"
+                        placeholder="Enter Your name"
+                        {...register("userName", {
+                          required: "Username is required",
+                          pattern: {
+                            value: /^[a-zA-Z]+[0-9]+$/,
+                            message:
+                              "Username must contain characters and end with numbers without spaces",
+                          },
+                          maxLength: { value: 8, message: "Username exceed 8" },
+                        })}
+                      />
+                      {errors?.userName && (
+                        <div className="alert alert-danger p-2 my-3">
+                          {errors?.userName?.message?.toString()}
+                        </div>
+                      )}
+                    </div>
+                  </div>
+                  <div className="col-md-6 ">
+                    <div className="input m-3">
+                      <label className="my-1">Email</label>
+                      <input
+                        type="email"
+                        className="form-control p-0"
+                        placeholder="Enter Your E-mail"
+                        {...register("email", {
```

Comment:
Email, password and other validation will be used in multiple places. I consider making a file containing all needed validations and then use this validation in our application. This will make our code more concise.

You can look at this example:
https://github.com/AnasKhaled7/PMS-React/blob/main/src/lib/InputValidator.ts